src/test/ui/parser/require-parens-for-chained-comparison.stderr

```diff
@@ -25,11 +25,20 @@ error: chained comparison operators require parentheses
    |
 LL |     f<Result<Option<X>, Option<Option<X>>>(1, 2);
    |      ^^^^^^^^
-help: use `::<...>` instead of `<...>` if you meant to specify type arguments
+help: use the "turbofish" `::<...>` instead of `<...>` to specify type arguments
    |
 LL |     f::<Result<Option<X>, Option<Option<X>>>(1, 2);
    |      ^^
 
+error: chained comparison operators require parentheses
+  --> $DIR/require-parens-for-chained-comparison.rs:22:21
+   |
+LL |     let _ = identity<u8>;
+   |                     ^^^^
+   |
+   = help: use the "turbofish" `::<...>` instead of `<...>` to specify type arguments
+   = help: or use `(...)` if you meant to specify fn arguments
+
```

Comment:
Possible heuristic: look-ahead to see if we have `;` -- at that point we can be sure it should be turbofish.

Reply:
This fails with `let _: Box<('a) + Trait>;`.

Reply:
But that's in a type context not expression context?